server/bin/pbench-server-put-shim-001.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+# This script is used to prepare the tarballs that a version 002 client
+# submits for further processing. It copies the tarballs and their MD5
+# sums to the archive (after checking) and sets the state links, so
+# that the dispatch script will pick them up and get the ball
+# rolling. IOW, it does impedance matching between version 002 clients
+# and the server scripts.
```

Comment:
Riya, unless you know what "impedance matching" is, I wouldn't use that analogy here.  (Actually, I don't think I would use it even if you do know what it is!  :-D)

Reply:
Sorry about that. I just copied the code ;). I actually don't know what it means will read about it :D

Reply:
LOL, unless you have an electrical engineering background, or you are an audiophile, or you are overhwhelmingly curious, I wouldn't bother reading about it!  :-D

All it really means is that you want the requirements of one system's or module's outputs to match the input requirements of the system or module that it's connected to, which would be pretty obvious to anyone with any situational awareness.  A problem arises when you have two systems or modules which you want to connect but whose requirements don't match -- there you have to insert a module between them which modifies the electrical circuit to make the respective impedances match, so that neither of them burns up.  (This sort of thing is most prevalent when connecting analog microphones to amplifiers, amplifiers to speakers, or other audio system components; hence my "audiophile" reference, above.)

As for this code, I would just remove the statement.  (If you really feel like we need to say something like this, I find that "glue" makes for a better analogy.  A "shim" is also an excellent analogy, but we've already used that in the name of this thing.  :-) )

Reply:
Thanks for that :D